src/mumble/pluginAPI/PluginComponents.h

```diff
@@ -0,0 +1,133 @@
+/**
+ * This header file contains definitions of types and other components used in Mumble's plugin system
+ */
+
+#ifndef MUMBLE_PLUGINCOMPONENT_H_
+#define MUMBLE_PLUGINCOMPONENT_H_
+
+#include <stdint.h>
+
+#define STATUS_OK 0
+
+struct Version {
+	int32_t major;
+	int32_t minor;
+	int32_t patch;
+};
+
+
+struct MumbleUser {
+	const char *name;
+	uint32_t id;
+};
+
+struct MumbleChannel {
+	const char *name;
+	int32_t id;
+};
+
+
+typedef struct Version Version_t;
+typedef struct MumbleUser MumbleUser_t;
+typedef struct MumbleChannel MumbleChannel_t;
+
+typedef int32_t MumbleConnection_t
+
+
+struct MumbleAPI {
+	// -------- Memory (de-)allocation --------
+	
+	/// Allocates memory for a MumbleUser. Once the user struct is no longer needed, it has to be freed by calling
+	/// freeMumbleUser.
+	///
+	/// @returns A pointer to the allocated struct.
+	MumbleUser_t* (*allocateMumbleUser)();
+
+	/// Frees a MumbleUser that has been allocated via allocateMumbleUser.
+	///
+	/// @params user A pointer to the user struct that should be freed
+	void (*freeMumbleUser)(MumbleUser_t *user);
+
+
+	/// Allocates memory for a MumbleChannel. Once the channel struct is no longer needed, it has to be freed by calling
+	/// freeMumbleChannel.
+	///
+	/// @returns A pointer to the allocated struct.
+	MumbleChannel_t* (*allocateMumbleChannel)();
+
+	/// Frees a MumbleChannel that has been allocated via allocateMumbleChannel.
+	///
+	/// @param channel A pointer to the channel struct that should get freed
+	void (*freeMumbleChannel)(MumbleChannel_t *channel);
+
+	
+
+	
+	// -------- Getter functions --------
+
+	/// Fills in the connection ID of the server the user is currently active on (the user's audio output is directed at).
+	///
+	/// @param[out] connection A pointer to the memory location the ID should be written to
+	/// @returns The error code. If everything went well, STATUS_OK will be returned. Only then it is valid to access the
+	/// 	value of the provided pointer
+	uint32_t getActiveServerConnection(MumbleConnection_t *connection);
+
+	/// Fills in the information about the local user.
+	///
+	/// @param connection A reference to which server-connection should be considered.
+	/// @param[out] user The user struct allocated via allocateMumbleUser whose fields will be set by this function.
+	/// @returns The error code. If everything went well, STATUS_OK will be returned. Only then the fields of the
+	/// 	passed struct may be accessed.
+	uint32_t (*getLocalUser)(MumbleConnection_t connection, MumbleUser_t *user);
```

Comment:
I'd say just return the id of the user, not a struct. We can then use other API to get name, current channel, ip ,whatever else we want